### PR TITLE
fix Issue 20497 - thread with limited stackspace crashes depending on…

### PR DESCRIPTION
**Pull request by rainers**
… size of TLS

on Posix systems, add the TLS size to the requested stack size for new threads

related issues:
- https://github.com/ldc-developers/ldc/issues/3127
- https://github.com/dlang/dub/issues/1812

### CODE REVIEW

**Comment by rainers** [comment]
The output of the test will hopefully show what platforms are actually using the stack for TLS.

File: src/core/thread/osthread.d [hunk 1]
```diff
@@ -3189,6 +3179,23 @@ private void onThreadError(string msg) nothrow @nogc
...
 
+version (Posix)
+private size_t adjustStackSize(size_t sz) nothrow @nogc
+{
+    if (sz == 0)
+        return 0;
+
+    // TLS uses the top of the stack, so add its size to the requested size
+    sz += externDFunc!("rt.sections_elf_shared.sizeOfTLS",
+                       size_t function() @nogc nothrow)();
```
Comment (kubo39) [inline src/core/thread/osthread.d]: Isn't it better to add `version(CRuntime_Glibc)` here?

**Comment by kubo39** [comment]
Nit: In glibc, we can use `__pthread_get_minstack` which is added in 2.15.

File: src/core/thread/osthread.d (hunk 1 above)
Comment (rainers) [reply src/core/thread/osthread.d]: Probably, as FreeBsd doesn't seem to have this issue. I've added the version check.

**Comment by rainers** [comment]
> Nit: In glibc, we can use `__pthread_get_minstack` which is added in 2.15.

Indeed, looks like a simpler solution. When searching for it I noticed a couple of bug reports due to undefined references to this symbol. Would it raise the libc version requirements for druntime?

**Pull request merged**